src/Build/Graph/ProjectGraphNode.cs

```diff
@@ -21,11 +23,16 @@ internal ProjectGraphNode(Project project)
         /// <summary>
         /// Gets an unordered collection of graph nodes for projects which this project references.
         /// </summary>
-        public IReadOnlyCollection<ProjectGraphNode> ProjectReferences => Array.Empty<ProjectGraphNode>();
+        public IReadOnlyCollection<ProjectGraphNode> ProjectReferences => _projectReferences;
 
         /// <summary>
         /// Gets the evaluated project represented by this node in the graph.
         /// </summary>
         public Project Project { get; }
+
+        internal void AddProjectReference(ProjectGraphNode projectGraphNode)
+        {
+            _projectReferences.Add(projectGraphNode);
```

Comment:
Nit: expression body since this method is so siple:
```
internal void AddProjectReference(ProjectGraphNode projectGraphNode) => _projectReferences.Add(projectGraphNode);
```